src/Tools/dotnet-monitor/Egress/Extension/WellKnownEgressExtensionRepository.cs

```diff
@@ -0,0 +1,33 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using Microsoft.Diagnostics.Tools.Monitor.Egress.FileSystem;
+using Microsoft.Diagnostics.Tools.Monitor.Extensibility;
+using Microsoft.Extensions.Logging;
+
+namespace Microsoft.Diagnostics.Tools.Monitor.Egress.Extension
+{
+    internal sealed class WellKnownEgressExtensionRepository :
+        ExtensionRepository
+    {
+        private readonly ILogger<FileSystemEgressExtension> _logger;
+
+        public WellKnownEgressExtensionRepository(ILogger<FileSystemEgressExtension> logger)
```

Comment:
Just confirming my understanding - a "Well Known" egress extension is just one that's built into dotnet-monitor - which in this case is just `FileSystem`? The only thing that feels a bit odd here is that the logger is specifically for `FileSystemEgressExtension`, even though the method is presumably meant to be generic, though for now that doesn't really matter since we're only looking for `FileSystem`.

Reply:
Yes, it's the only statically known egress extension because it's "in the box". Removing the `ILogger` injection would mean that we have some other implementation class that gets injected that would create the `FileSystemEgressExtension` instance, which is basically the same problem.

I'll play around with is some more to see if I can remove the dependency on the concrete types.